### PR TITLE
Add crossLanguageDefinitionId to apiview-properties.json

### PR DESCRIPTION
**Pull request by srnagar**

### CODE REVIEW

File: typespec-extension/src/common/operation.ts [hunk 1]
```diff
@@ -153,6 +155,10 @@ export class Operation extends Aspect implements Operation {
     (this.profile = this.profile || {})[profileName] = apiVersion;
     return this;
   }
+
+  setCrossLanguageDefinitionId(id: string) {
+    this.crossLanguageDefinitionId = id;
+  }
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/common/operation.ts:161]: I think typescript usually does not need a setter.

File: typespec-tests/main.tsp
Comment (weidongxu-microsoft) [inline typespec-tests/main.tsp:1]: this file should not be committed?

File: typespec-extension/src/code-model-builder.ts
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:1]: please run `npm run format` under typespec-extension

File: typespec-extension/src/code-model-builder.ts [hunk 2]
```diff
@@ -1730,37 +1733,37 @@ export class CodeModelBuilder {
...
+            namespace: namespace,
           },
-        }),
-      );
+          java: {
+            namespace: getJavaNamespace(namespace),
+          },
+        },
+      });
+      sealedChoiceSchema.crossLanguageDefinitionId = getCrossLanguageDefinitionId(type);
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:1749]: Totally perisomal opinion and likely very subjective: I kind of inclined to put `crossLanguageDefinitionId` into `language.default.crossLanguageDefinitionId` so it applies to any code model object (`crossLanguageDefinitionId` being a pretty universal identity, like `namespace`+`name` and `description` etc.).

No need to change, we can discuss and adjust in follow-up PR.

File: typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java [hunk 3]
```diff
@@ -48,6 +53,47 @@ public Client processClient(CodeModel codeModel) {
         // map to client model
         Client client = Mappers.getClientMapper().map(codeModel);
 
+        client.getAsyncClients()
+                .forEach(asyncClient -> crossLanguageDefinitionsMap
+                        .put(asyncClient.getPackageName() + "." + asyncClient.getClassName(), asyncClient.getCrossLanguageDefinitionId()));
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java:58]: Is it possible that some of these `.getCrossLanguageDefinitionId()` returns `null`?

If we just output `null` in `apiview_properties.json`, then we are good.

Comment (srnagar) [reply typespec-extension/src/main/java/com/azure/autorest/TypeSpecPlugin.java:55]: Yes, the json will have null. Not a big issue as apiview is able to handle it.

**Review by weidongxu-microsoft — APPROVED** [review]
@XiaofeiCao Please also take a look on whether this may impact mgmt.

**Pull request merged**